### PR TITLE
fix: Bff 646 download yearly overview

### PR DESCRIPTION
**Pull request by Daniela510**
## Description

Don't merge yet!  
I need to add github env variables for aws base api.

This PR switches to the newer AWS endpoint. 
Now the yearly summary comes as a separate PDF in email.
Also reverted the overview page to being state-less.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/features/overview/bloc/givt_bloc.dart [hunk 1]
```diff
@@ -233,9 +233,11 @@ class GivtBloc extends Bloc<GivtEvent, GivtState> {
     Emitter<GivtState> emit,
   ) async {
     try {
+      final fromDate = '${event.year}-01-01';
+      final tillDate = '${event.year}-12-31';
+
```
Comment (alexbejann) [inline lib/features/overview/bloc/givt_bloc.dart:238]: Maybe is better to use a DateTime object and a DateFormatter

File: lib/features/overview/bloc/givt_bloc.dart [hunk 2]
```diff
@@ -244,14 +246,10 @@ class GivtBloc extends Bloc<GivtEvent, GivtState> {
...
-      await LoggingInfo.instance.error(
-        e.toString(),
-        methodName: StackTrace.current.toString(),
-      );
-      emit(GivtError(
-        e.toString(),
-      ));
+    } on GivtServerFailure catch (e) {
+      await LoggingInfo.instance.info(e.body.toString());
+      await LoggingInfo.instance.error(e.body.toString());
```
Comment (alexbejann) [inline lib/features/overview/bloc/givt_bloc.dart:251]: Do you need both?

File: .github/workflows/android.yml [hunk 3]
```diff
@@ -27,6 +27,8 @@ jobs:
             echo "AMPLITUDE_KEY=f4fa9ab88de04c56a346aaa36c172a9a" >> $GITHUB_ENV
             echo "API_URL_US=backend.givt.app" >> $GITHUB_ENV
             echo "API_URL_EU=backend.givtapp.net" >> $GITHUB_ENV
+            echo "API_URL_AWS_US=api.production.givt.app" >> $GITHUB_ENV
+            echo "API_URL_AWS_EU=api.production.givtapp.net" >> $GITHUB_ENV
```
Comment (alexbejann) [inline .github/workflows/android.yml:31]: This would have to be cleaned once we get rid of the AWS stuff

**Review by alexbejann — APPROVED** [review]
Just some remarks, but it's good enough

File: lib/features/overview/bloc/givt_bloc.dart (hunk 1 above)
Comment (Daniela510) [reply lib/features/overview/bloc/givt_bloc.dart:238]: To use DateTime and a DateFormatter the lines would look like:
```
final fromDate = DateFormat('yyyy-MM-dd').format(DateTime.utc(int.parse(year), 01, 01));
final tillDate = DateFormat('yyyy-MM-dd').format(DateTime.utc(int.parse(year), 12, 31));
```
I dont really see the point of parsing the year string into and integer, to then create a dateTime object, to then format that into a string again. 
What are the benefits of using DateTime?

File: lib/features/overview/bloc/givt_bloc.dart (hunk 2 above)
Comment (Daniela510) [reply lib/features/overview/bloc/givt_bloc.dart:251]: Nope

**Pull request merged**